x-pack/legacy/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -361,11 +361,30 @@ export function XYChart({
             enableHistogramMode: isHistogram && (seriesType.includes('stacked') || !splitAccessor),
             timeZone,
             name(d) {
+              const splitHint = table.columns.find(col => col.id === splitAccessor)?.formatHint;
+
+              // For multiple y series, the name of the operation is used on each, either:
+              // * Key - Y name
+              // * Formatted value - Y name
               if (accessors.length > 1) {
                 return d.seriesKeys
-                  .map((key: string | number) => columnToLabelMap[key] || key)
+                  .map((key: string | number, i) => {
+                    if (i === 0 && splitHint) {
+                      return formatFactory(splitHint).convert(key);
+                    }
+                    return columnToLabelMap[key] ?? key;
```

Comment:
Yes, the empty string is possible if the user deletes the default label that we apply. It could be more correct in that case to use the logic `columnLabelMap[key] || key` which will fall back to showing the key.

Reply:
I've now updated this logic so that it's impossible to see the UUIDs that we use internally